src/tests/ftest/launch.py

```diff
@@ -276,141 +315,156 @@ def get_available_interfaces(args):
     """Set up the provider environment variables.
 
     Use the existing CRT_PHY_ADDR_STR setting if already defined, otherwise
     select the appropriate provider based upon the interface driver.
 
     Args:
+        log (logger): logger for the messages produced by this method
         interface (str): the current interface being used.
+        args (argparse.Namespace): command line arguments for this program
+
+    Raises:
+        LaunchException: if there is a problem finding a provider for the interface
+
     """
+    log.debug("-" * 80)
     # Use the detected provider if one is not set
     if args.provider:
         provider = args.provider
     else:
         provider = os.environ.get("CRT_PHY_ADDR_STR")
-    if provider is None:
-        log("Detecting provider for {} - CRT_PHY_ADDR_STR not set".format(interface))
+    if provider is None:    # pylint: disable=too-many-nested-blocks
+        log.debug("Detecting provider for %s - CRT_PHY_ADDR_STR not set", interface)
 
         # Check for a Omni-Path interface
+        log.debug("Checking for Omni-Path devices")
         command = "sudo opainfo"
-        task = get_remote_output(args.test_servers, command)
-        if check_remote_output(task, command):
+        result = run_remote(log, args.test_servers, command)
+        if result.passed:
             # Omni-Path adapter not found; remove verbs as it will not work with OPA devices.
-            log("  Excluding verbs provider for Omni-Path adapters")
+            log.debug("  Excluding verbs provider for Omni-Path adapters")
             PROVIDER_KEYS.pop("verbs")
 
         # Detect all supported providers
-        command = "fi_info -d {} -l | grep -v 'version:'".format(interface)
-        task = get_remote_output(args.test_servers, command)
-        if check_remote_output(task, command):
-            # Verify each server host has the same interface driver
-            output_data = list(task.iter_buffers())
-            if len(output_data) > 1:
-                log("ERROR: Non-homogeneous drivers detected.")
-                sys.exit(1)
-
+        command = f"fi_info -d {interface} -l | grep -v 'version:'"
+        result = run_remote(log, args.test_servers, command)
+        if result.passed:
             # Find all supported providers
-            keys_found = []
-            for line in output_data[0][0]:
-                provider_name = line.decode("utf-8").replace(":", "")
-                if provider_name in PROVIDER_KEYS:
-                    keys_found.append(provider_name)
+            keys_found = {}
+            for data in result.output:
+                for line in data.stdout:
+                    provider_name = line.replace(":", "")
+                    if provider_name in PROVIDER_KEYS:
+                        if provider_name not in keys_found:
+                            keys_found[provider_name] = NodeSet()
+                        keys_found[provider_name].update(data.hosts)
```

Comment:
If we wanted to circumvent the `pylint: disable=too-many-nested-blocks`, this will remove two levels
```suggestion
            from collections import defaultdict
            keys_found = defaultdict(NodeSet)
            for line in (data.stdout for data in result.output):
                provider_name = line.replace(":", "")
                if provider_name in PROVIDER_KEYS:
                    keys_found[provider_name].update(data.hosts)
```

Reply:
Thanks for the suggestion, but `data` is inaccessible when trying to call `update(data.hosts)`.  As it turns out I don't need the `pylint: disable=too-many-nested-blocks` anyway as our pylint.rc file allows for a larger number than the default.